MiniApp/Classes/MiniAppClient.swift

```diff
@@ -26,17 +28,52 @@ class MiniAppClient {
             switch result {
             case .success(let responseData):
                 if !(200...299).contains(responseData.httpResponse.statusCode) {
-                    do {
-                        let errorModel = try JSONDecoder().decode(ErrorData.self, from: responseData.data)
-                        return completionHandler(.failure(NSError.serverError(code: errorModel.code, message: errorModel.message)))
-                    } catch {
-                        return completionHandler(.failure(NSError.unknownServerError(httpResponse: responseData.httpResponse)))
-                    }
+                    return completionHandler(.failure(
+                        self.handleHttpResponse(responseData: responseData.data,
+                                                httpResponse: responseData.httpResponse)
+                    ))
                 }
-                return completionHandler(.success(ResponseData(responseData.data, responseData.httpResponse)))
+                return completionHandler(.success(ResponseData(responseData.data,
+                                                               responseData.httpResponse)))
             case .failure(let error):
                 return completionHandler(.failure(error))
             }
         }
     }
+
+    func getAppManifest(appId: String,
+                        versionId: String,
+                        completionHandler: @escaping (Result<ManifestResponse, Error>) -> Void) {
+
+        guard let urlRequest = self.manifestApi.createURLRequest(appId: appId, versionId: versionId) else {
+            return completionHandler(.failure(NSError.invalidURLError()))
+        }
+
+        session.startDataTask(with: urlRequest) { (result) in
+            switch result {
+            case .success(let responseData):
+                if !(200...299).contains(responseData.httpResponse.statusCode) {
```

Comment:
It looks like this `if` statement and its contents are the same in `getMiniAppsList`. Is there any way to reduce the code duplication? Maybe everything in the `.success` case could be pulled out into some kind of generic `handleSuccessResponse` function?

Reply:
Done